src/modules/bootloader/main.py

```diff
@@ -637,28 +637,33 @@ def run_grub_install(fw_type, partitions, efi_directory):
                                    "--bootloader-id=" + efi_bootloader_id,
                                    "--force"])
     else:
-        assert efi_directory is None
-        if libcalamares.globalstorage.value("bootLoader") is None:
-            return
+        if efi_directory is not None:
+            if not install_hybrid_grub:
+                return
 
-        boot_loader = libcalamares.globalstorage.value("bootLoader")
-        if boot_loader["installPath"] is None:
-            return
+        if libcalamares.globalstorage.value("bootLoader") is None:
+            find_esp_disk_command = "echo \"/dev/$(lsblk -o PKNAME $(df --output=source /boot/efi | tail -n1) | tail -n1)\""
+            boot_loader_install_path = check_target_env_output(["sh", "-c", find_esp_disk_command]).strip()
```

Comment:
We should probably check to ensure that `boot_loader_install_path` is something like we expect.

Reply:
Added a check to make sure the path begins with `/dev/` and is at least eight characters long (the length of `/dev/sda`, to my awareness there aren't any two-character disk names in Linux).